### PR TITLE
Bug fix in PathFinder

### PR DESCRIPTION
**Pull request by jmmshn**
The way PathFinder figures out the `iindex` and `eindex`
The code that is breaking is
https://github.com/materialsvirtuallab/pymatgen-diffusion/blob/master/pymatgen_diffusion/neb/pathfinder.py#L301

where the code is trying to match the `esite` to only the first instance of the site in the symmetry group.
Now the code checks all possible sites if the initial symmetry matching fails.

- some other small cleanup and linting

### CODE REVIEW

File: pymatgen_diffusion/neb/full_path_mapper.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def __init__(
         # Generate the graph edges between these all the sites
         self.s_graph = StructureGraph.with_local_env_strategy(
             self.only_sites,
-            MinimumDistanceNN(tol=np.infty, cutoff=max_path_length, get_all_sites=True),
+            MinimumDistanceNN(cutoff=max_path_length, get_all_sites=True),
```
Comment (jmmshn) [inline pymatgen_diffusion/neb/full_path_mapper.py:130]: The first parameter is not needed if `get_all_sites=True` is set

File: pymatgen_diffusion/neb/pathfinder.py [hunk 2]
```diff
@@ -290,25 +312,68 @@ def __init__(self, isite, esite, symm_structure):
...
+            esite.specie, (isite.frac_coords + esite.frac_coords) / 2, esite.lattice
+        )
         sg = self.symm_structure.spacegroup
         for i, sites in enumerate(self.symm_structure.equivalent_sites):
             if sg.are_symmetrically_equivalent([isite], [sites[0]]):
                 self.iindex = i
             if sg.are_symmetrically_equivalent([esite], [sites[0]]):
                 self.eindex = i
 
+        # if no index was identified then loop over each site until something is found
```
Comment (jmmshn) [inline pymatgen_diffusion/neb/pathfinder.py:328]: All the new logic is in this block everything else is just reformatting by pre-commit.yaml + black

File: pymatgen_diffusion/neb/tests/test_full_path_mapper.py [hunk 3]
```diff
@@ -178,6 +177,12 @@ def test_get_intercalating_path(self):
         }
         self.assertIn("1->0->1", p_strings)
 
+    def test_not_matching_first(self):
```
Comment (jmmshn) [inline pymatgen_diffusion/neb/tests/test_full_path_mapper.py:180]: This was a case that could not populate the index before.

File: setup.cfg [hunk 4]
```diff
@@ -8,5 +8,5 @@ exclude=test_*
 [flake8]
 exclude = .git,__pycache__,docs_rst/conf.py,tests,__init__.py
 # max-complexity = 10
-extend-ignore = E741,W291,W293,E501,E231,E203
+extend-ignore = E741,W291,W293,E501,E231,E203,W605
```
Comment (jmmshn) [inline setup.cfg:11]: Flake 8 for `\*\*` escape characters

**Review by jmmshn — COMMENTED** [review]
Added comments to highlight changes

**Comment by shyuep** [comment]
Thanks.

**Pull request merged**